src/wp-admin/includes/class-wp-upgrader.php

```diff
@@ -593,8 +594,20 @@ public function install_package( $args = array() ) {
 			}
 		}
 
-		// Copy new version of item into place.
-		$result = copy_dir( $source, $remote_destination );
+		/*
+		 * Partial updates may want to retain the destination.
+		 * move_dir() returns a WP_Error when the destination exists,
+		 * so copy_dir() should be used.
+		 *
+		 * If 'clear_working' is false, the source shouldn't be removed.
+		 * After move_dir() runs, the source will no longer exist.
+		 * Therefore, copy_dir() should be used.
+		 */
+		if ( $clear_destination && $args['clear_working'] ) {
+			$result = move_dir( $source, $remote_destination );
```

Comment:
```suggestion
			$result = move_dir( $source, $remote_destination );
			$wp_filesystem->delete( $remote_source, true );
```

Agree with Peter, needs to remove https://github.com/WordPress/wordpress-develop/blob/3b97eac7793078005adc552a985ebd58f526fecc/src/wp-admin/includes/class-wp-upgrader.php#L612-L615

Reply:
@mukeshpanchal27 there has been much discussion in several places since Peter wrote the above. Additionally the the PR has changed. The original suggestion by Peter has changed from what it is currently

That bit of code you wish to modify/remove has been in place for over 8 years and to change it to your suggestion is to essentially modify it to as follows by putting it in the preceding conditional.

```php
if ( $clear_destination && $args['clear_working'] ) {
      $wp_filesystem->delete( $remote_source, true ); 
 } 
```

The implication is that this has been incorrect the entire time. I don't believe you realized that the PR has been updated since Peter made the above suggestion.

Reply:
Thanks @afragen I miss the updates. Your suggestion look good to me. thanks.